### PR TITLE
Adds the Death Sandwich to the game

### PR DESCRIPTION
**Pull request by Wallemations**
## About The Pull Request
Adds the Death Sandwich to the game, the ultimate form of bread-conveyed-meat-based consumables.
deathsandwich

And remember;
Eat it right, or you die!
## Why It's Good For The Game
I'm genuinely surprised we don't already have a meatball sub in the game
also I love humor food, and I doubt my edition of the Eggcellent Challenge would ever be merged if I tried to do so, so this is the next best thing.
## Changelog
:cl: Wallem
add: The ancient recipe for the Death Sandwich has been rediscovered buried in the deepest depths of an erupting volcano.
/:cl:

### CODE REVIEW

**Comment by carshalash** [comment]
I'd say replace the cutlets in the recipe with salami.

**Comment by Wallemations** [comment]
> I'd say replace the cutlets in the recipe with salami.

We have salami in our codebase?

**Comment by carshalash** [comment]
> > I'd say replace the cutlets in the recipe with salami.
> 
> We have salami in our codebase?

By slicing a sausage with a knife, you are able to produce 6 salami slices.

**Comment by carshalash** [comment]
If you want, I'd recommend adding a mute for a short duration. Fits in with the whole vocal loss part of the later parts of the sandwich.

**Comment by Wallemations** [comment]
> If you want, I'd recommend adding a mute for a short duration. Fits in with the whole vocal loss part of the later parts of the sandwich.

emoji

**Comment by carshalash** [comment]
Based PR

**Review by Cyberboss — APPROVED** [review]

**Comment by Orchian** [comment]
Based.

File: code/game/objects/items/food/sandwichtoast.dm [hunk 1]
```diff
@@ -241,3 +241,34 @@
...
+	food_reagents = list(
+		/datum/reagent/consumable/nutriment = 8,
+		/datum/reagent/consumable/nutriment/protein = 14,
+		/datum/reagent/consumable/nutriment/vitamin = 6,
+	)
+	tastes = list("bread" = 1, "meat" = 1, "tomato sauce" = 1, "death" = 1)
+	foodtypes = GRAIN | MEAT
+	food_flags = FOOD_FINGER_FOOD
+	w_class = WEIGHT_CLASS_SMALL
+	eat_time = 4 SECONDS // I THINK this is longer than normal? Makes it harder to force-feed this to people as a weapon, as funny as that is.
```
Comment (san7890) [inline code/game/objects/items/food/sandwichtoast.dm:259]: i think having an `eat_time` like this is an _okay_ safeguard, but we should really do the arnold pizza thing where forcefeeding it doesn't work.

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/food/sandwichtoast.dm (hunk 1 above)
Comment (Wallemations) [reply code/game/objects/items/food/sandwichtoast.dm:259]: arnold pizza has that? I'll have to see how it does that when I wake up tomorrow.

File: code/game/objects/items/food/sandwichtoast.dm [hunk 2]
```diff
@@ -241,3 +241,34 @@
...
+
+///Eat it right, or you die.
+/obj/item/food/death_sandwich/proc/check_liked(fraction, mob/living/carbon/human/consumer)
+	/// Closest thing to a mullet we have
+	if(consumer.hairstyle == "Gelled Back" && istype(consumer.get_item_by_slot(ITEM_SLOT_ICLOTHING), /obj/item/clothing/under/rank/civilian/cookjorts))
+		return FOOD_LIKED
+	// I thought it didn't make sense for it to instantly kill you, so instead enjoy shitloads of toxin damage per bite.
+	balloon_alert(consumer, "ate it wrong!")
+	consumer.adjustToxLoss(50)
+	consumer.set_silence_if_lower(5 SECONDS)
```
Comment (LemonInTheDark) [inline code/game/objects/items/food/sandwichtoast.dm:274]: Does this deserve audio feedback? Seems like a chance for soul

**Comment by CRITAWAKETS** [comment]
that sprite needs an extra shade on the bread

**Comment by IndieanaJones** [comment]
Do we have jorts? Where do you get them?

File: code/game/objects/items/food/sandwichtoast.dm (hunk 2 above)
Comment (Wallemations) [reply code/game/objects/items/food/sandwichtoast.dm:274]: What kinda feedback we talking?

File: code/game/objects/items/food/sandwichtoast.dm [hunk 3]
```diff
@@ -241,3 +241,34 @@
...
+/obj/item/food/death_sandwich/make_edible()
+	. = ..()
+	AddComponent(/datum/component/edible, check_liked = CALLBACK(src, PROC_REF(check_liked)))
+
+///Eat it right, or you die.
+/obj/item/food/death_sandwich/proc/check_liked(fraction, mob/living/carbon/human/consumer)
+	/// Closest thing to a mullet we have
+	if(consumer.hairstyle == "Gelled Back" && istype(consumer.get_item_by_slot(ITEM_SLOT_ICLOTHING), /obj/item/clothing/under/rank/civilian/cookjorts))
+		return FOOD_LIKED
+	// I thought it didn't make sense for it to instantly kill you, so instead enjoy shitloads of toxin damage per bite.
```
Comment (Y0SH1M4S73R) [inline code/game/objects/items/food/sandwichtoast.dm:271]: In the source material, if you eat it wrong, it takes six hours to die, with the symptoms occurring gradually over that time frame. So maybe instead code those symptoms to occur over the course 1/4th of a spessmen day/night cycle?

Comment (Y0SH1M4S73R) [reply code/game/objects/items/food/sandwichtoast.dm:271]: It's the **double** death sandwich that is not just an instant death, but occurs regardless of being eaten correctly.

File: code/game/objects/items/food/sandwichtoast.dm (hunk 1 above)
Comment (san7890) [reply code/game/objects/items/food/sandwichtoast.dm:259]: last i remember: yes

File: code/game/objects/items/food/sandwichtoast.dm (hunk 3 above)
Comment (necromanceranne) [reply code/game/objects/items/food/sandwichtoast.dm:271]: maybe a non-transferable disease?

Comment (Y0SH1M4S73R) [reply code/game/objects/items/food/sandwichtoast.dm:271]: sounds about right. Call it "death sandwich poisoning" or something and have the cure be the corresponding sandwich of life.

File: code/game/objects/items/food/sandwichtoast.dm (hunk 1 above)
Comment (Fikou) [reply code/game/objects/items/food/sandwichtoast.dm:259]: i dont like that idea, if you eat it impropery you get the pain. itd be stupid if you could just be forcefed it

**Comment by Overuseddart** [comment]
Pneumatic cannon would force someone to eat this instantly, like a bombger!

**Comment by celulamp** [comment]
Should have added the broodwich

File: code/game/objects/items/food/sandwichtoast.dm (hunk 1 above)
Comment (Wallemations) [reply code/game/objects/items/food/sandwichtoast.dm]: I thought you meant that Arnold pizza simply doesn't let you forcefeed other people, the fact that it just bypasses the check and that's it makes me want to keep it the way I did it instead.

**Comment by tralezab** [comment]
I kinda just wish we had a normal meatball sub. But that being said, I like death sandwich. I would just make it a bit more costly crafting-wise.

**Comment by tralezab** [comment]
> Should have added the broodwich
> 
> and it teleports you to lavaland if you eat it

LOL

**Comment by Wallemations** [comment]
> I kinda just wish we had a normal meatball sub. But that being said, I like death sandwich. I would just make it a bit more costly crafting-wise.

I always took part of the joke of the Death Sandwich to be the fact that visually, the guy's just making a normal meatball sub more or less.
Also I'm pretty sure if I added more ingredients it would end up having WAY too many reagents inside it.

**Comment by tralezab** [comment]
Yeah but maybe I wanna serve a meatball sub without killing someone :9

**Comment by Fikou** [comment]
this could be an unlockable recipe thatd be cool

**Comment by Wallemations** [comment]
> this could be an unlockable recipe thatd be cool

Unlockable how?

**Comment by necromanceranne** [comment]
You can get crafting recipe granters. Stuff like the cookbooks in the chef vendors.

**Comment by Shadowflame909** [comment]
Clown 1 death sandwich and 1 Arnold pizza please

Medical Doctors will have their hands tied with this one. Great PR!

**Comment by Wallemations** [comment]
I'm not adding anything else to this PR, it's already bikeshedding a lot as it is.

**Pull request merged**